### PR TITLE
Return count when updating or deleting models

### PR DESCRIPTION
**Pull request by superkhau**
Connect to strongloop/loopback#1167

### CODE REVIEW

**Comment by superkhau** [comment]
@raymondfeng PTAL

/cc @bajtos

File: lib/mssql.js [hunk 1]
```diff
@@ -318,12 +318,11 @@ MsSQL.prototype.destroyAll = function (model, where, cb) {
...
-  this.query('DELETE FROM '
-    + this.tableEscaped(model) + ' ' + this.buildWhere(model, where || {}),
-    function (err, data) {
-    cb && cb(err, data);
-  }.bind(this));
-
+  var sql = 'DELETE FROM ' + this.tableEscaped(model) + ' ' +
+      this.buildWhere(model, where || {}) + ';SELECT @@ROWCOUNT as count';
+  this.query(sql, function(err, info) {
+    cb && cb(err, info[0]);
```
Comment (bajtos) [inline lib/mssql.js]: Is this assuming that `info[0]` is an object with a `count` property? I find it a bit cryptic, can you perhaps add a comment to explain?

File: lib/mssql.js [hunk 2]
```diff
@@ -365,9 +364,12 @@ MsSQL.prototype.update = MsSQL.prototype.updateAll = function (model, where, dat
   sql = "UPDATE " + tblName + MsSQL.newline;
   sql += "SET " + combined.join() + MsSQL.newline;
   sql += self.buildWhere(model, where) + MsSQL.newline;
+  sql += ';SELECT @@ROWCOUNT as count';
   fieldValues.push(id);
 
-  self.query(sql, fieldValues, cb);
+  self.query(sql, fieldValues, function(err, info) {
+    cb(err, info[0]);
```
Comment (bajtos) [inline lib/mssql.js]: Is this assuming that `info[0]` is an object with a `count` property? I find it a bit cryptic, can you perhaps add a comment to explain?

File: lib/mssql.js (hunk 1 above)
Comment (raymondfeng) [inline lib/mssql.js]: You need to check if info exists. It will be undefined if err is reported.

File: lib/mssql.js (hunk 2 above)
Comment (raymondfeng) [inline lib/mssql.js]: same as above

**Comment by superkhau** [comment]
@raymondfeng PTAL again.

/cc @bajtos

**Comment by bajtos** [comment]
I don't see any glaring problem.

**Comment by bajtos** [comment]
@slnode test please

**Comment by bajtos** [comment]
FYI: the test `mssql imported features manipulation create.should refuse to create object with duplicate id	` is failing with `Could not drop object 'dbo.Product' because it is referenced by a FOREIGN KEY constraint.` While this is almost certainly not caused by this patch, I am wondering if we can trust results of other tests when one of the tests failed like this?

**Comment by raymondfeng** [comment]
LGTM

**Comment by superkhau** [comment]
@slnode test please

**Comment by superkhau** [comment]
@bajtos @raymondfeng Addressed the previous code review comments. Merging.

**Pull request merged**